src/devtools/server.ts

```diff
@@ -15,8 +15,53 @@ devtoolsEvents.addListener("fromDevTools", (msg) => {
   Debug.info("DevTools > WS: " + JSON.stringify(msg));
 });
 
-export function startServer(port: number): Disposable {
+let id = 1;
+
+export function sendToDevTools(message: unknown) {
+  devtoolsEvents.emit("toDevTools", {
+    id: `vscode-${id++}`,
+    source: "apollo-client-devtools",
+    type: "actor",
+    message,
+  });
+}
+
+export let serverState:
+  | { port: false | number; disposable: Disposable }
+  | undefined = undefined;
```

Comment:
`false` if the server is in the middle of starting, `number` if the server is running. The whole `serverState` is `undefined` after the server has stopped or if it has never been started before.

Reply:
It felt more deliberate than `undefined`. `null` would have been fine too, I guess.